--- core/include/cv32a6_imac_sv32_scaiev_config_pkg.sv
+++ core/include/cv32a6_imac_sv32_scaiev_config_pkg.sv
@@ -36,11 +36,11 @@ package cva6_config_pkg;
   localparam CVA6ConfigDataUserEn = 0;
   localparam CVA6ConfigDataUserWidth = CVA6ConfigXlen;
 
-  localparam CVA6ConfigIcacheByteSize = 16384;
-  localparam CVA6ConfigIcacheSetAssoc = 4;
+  localparam CVA6ConfigIcacheByteSize = 8192;
+  localparam CVA6ConfigIcacheSetAssoc = 2;
   localparam CVA6ConfigIcacheLineWidth = 128;
-  localparam CVA6ConfigDcacheByteSize = 16384;
-  localparam CVA6ConfigDcacheSetAssoc = 4;
+  localparam CVA6ConfigDcacheByteSize = 8192;
+  localparam CVA6ConfigDcacheSetAssoc = 2;
   localparam CVA6ConfigDcacheLineWidth = 128;
 
   localparam CVA6ConfigDcacheIdWidth = 1;
@@ -57,8 +57,8 @@ package cva6_config_pkg;
   localparam CVA6ConfigNrLoadBufEntries = 2;
 
   localparam CVA6ConfigRASDepth = 2;
-  localparam CVA6ConfigBTBEntries = 32;
-  localparam CVA6ConfigBHTEntries = 128;
+  localparam CVA6ConfigBTBEntries = 16;
+  localparam CVA6ConfigBHTEntries = 64;
 
   localparam CVA6ConfigTvalEn = 1;
 
--- core/include/cv64a6_imac_sv39_scaiev_config_pkg.sv
+++ core/include/cv64a6_imac_sv39_scaiev_config_pkg.sv
@@ -36,11 +36,11 @@ package cva6_config_pkg;
   localparam CVA6ConfigDataUserEn = 0;
   localparam CVA6ConfigDataUserWidth = CVA6ConfigXlen;
 
-  localparam CVA6ConfigIcacheByteSize = 16384;
-  localparam CVA6ConfigIcacheSetAssoc = 4;
+  localparam CVA6ConfigIcacheByteSize = 8192;
+  localparam CVA6ConfigIcacheSetAssoc = 2;
   localparam CVA6ConfigIcacheLineWidth = 128;
-  localparam CVA6ConfigDcacheByteSize = 16384;
-  localparam CVA6ConfigDcacheSetAssoc = 4;
+  localparam CVA6ConfigDcacheByteSize = 8192;
+  localparam CVA6ConfigDcacheSetAssoc = 2;
   localparam CVA6ConfigDcacheLineWidth = 128;
 
   localparam CVA6ConfigDcacheIdWidth = 1;
@@ -57,8 +57,8 @@ package cva6_config_pkg;
   localparam CVA6ConfigNrLoadBufEntries = 2;
 
   localparam CVA6ConfigRASDepth = 2;
-  localparam CVA6ConfigBTBEntries = 32;
-  localparam CVA6ConfigBHTEntries = 128;
+  localparam CVA6ConfigBTBEntries = 16;
+  localparam CVA6ConfigBHTEntries = 64;
 
   localparam CVA6ConfigTvalEn = 1;
 
