--- ip/orca/hdl/sys_call.vhd	2023-10-16 18:45:17.279963411 +0200
+++ ip/orca/hdl/sys_call.vhd	2023-10-16 18:45:57.551629376 +0200
@@ -420,7 +420,7 @@
           mstatus(CSR_MSTATUS_MPIE)       <= '0';
           mepc                            <= (others => '0');
           mcause(mcause'left)             <= '0';
-          mcause_exc_code                 <= (others => '0');
+          mcause_exc_code(CSR_MCAUSE_CODE'range) <= (others => '0');
           meimask_full                    <= (others => '0');
         end if;
         mepc(1 downto 0) <= "00";
